src/redux/slice.js

```diff
@@ -0,0 +1,38 @@
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
+
+const initialState = { data: [], loading: false, error: null };
+
+export const getPollutionData = createAsyncThunk(
+  'pollutionData',
+  async (url) => {
+    const data = await fetch(url);
+    const pollutionInfo = await data.json();
+    return pollutionInfo;
+  },
+);
+
+export const polutionSlice = createSlice({
+  name: 'counter',
+  initialState,
+  reducers: {},
+  extraReducers: (Builder) => {
+    Builder.addCase(getPollutionData.pending, (state) => {
+      const newState = state;
+      newState.loading = true;
+      newState.error = null;
+      newState.data = [];
+      return newState;
+    }).addCase(getPollutionData.fulfilled, (state, { payload }) => {
+      const newState = state;
+      newState.loading = false;
+      const arr = payload.list[0];
+      const components = Object.entries(arr.components);
+      newState.data = components;
+      return newState;
+    });
+  },
+});
+
+// export const { increment, decrement, incrementByAmount } = counterSlice.actions;
```

Comment:
- [ ] Kindly ensure that you remove all commented-out, un used/ un wanted pieces of code from your code base. It makes it look cluttered and messy.